`codecov/patch` failed at **33.33% of diff hit** and it was pointing at something real, so I fixed the gap rather than waive it.

The uncovered lines were the error-handling path in `ScriptLineStep.close()` - which is exactly the behaviour the Javadoc asserts ("release every line, rethrow the first failure") and which had **no test at all**. `SQLScriptLargeBatchTest` only drives the happy path.

Added `ScriptLineStepCloseTest` (bf453bac0): a chain of stub `InternalExecutionPlan`s, asserting both halves of the contract.

- every line is released tail to head
- a plan that throws does not strand the ones behind it, and the tail-most failure is the one that surfaces

I verified the test can fail rather than trusting a green run: the second case fails against the previous `try/finally` implementation, which let the head-most failure replace the original.

```
EXIT_ON_OLD_IMPL=1
[ERROR] ScriptLineStepCloseTest.aPlanThatFailsToCloseDoesNotStrandTheLinesBehindIt:124
```

The one line still uncovered is the `current != null` defensive tail, which is unreachable today for the reason documented in the comment - `chain()` is the only writer of a script line's previous step and always passes another script line. I would rather leave it uncovered than delete a guard or write a test that fakes an impossible chain shape.

Local run with the new test: `ScriptLineStepCloseTest`, `SQLScriptLargeBatchTest`, `SQLScriptTest`, `SQLScriptAdditionalCoverageTest`, `BatchTest` = 80 tests, 0 failures.

--- a/engine/src/test/java/com/arcadedb/query/sql/executor/ScriptLineStepCloseTest.java
+++ b/engine/src/test/java/com/arcadedb/query/sql/executor/ScriptLineStepCloseTest.java
@@ -81,6 +81,37 @@ public class ScriptLineStepCloseTest {
   }
 
   /**
+   * A step of a kind other than a script line. {@link ScriptExecutionPlan#chain} never puts one behind a
+   * script line, but the chain is typed on {@link ExecutionStepInternal}, so the walk guards against it.
+   */
+  private static class ForeignStep extends AbstractExecutionStep {
+    private final List<String>     closed;
+    private final RuntimeException failure;
+
+    ForeignStep(final CommandContext context, final List<String> closed, final RuntimeException failure) {
+      super(context);
+      this.closed = closed;
+      this.failure = failure;
+    }
+
+    @Override
+    public ResultSet syncPull(final CommandContext context, final int nRecords) {
+      return new InternalResultSet();
+    }
+
+    @Override
+    public void close() {
+      closed.add("foreign");
+      try {
+        if (failure != null)
+          throw failure;
+      } finally {
+        super.close();
+      }
+    }
+  }
+
+  /**
    * Builds a chain of script lines in script order and returns the tail, which is what
    * {@link ScriptExecutionPlan#close()} closes.
    */
@@ -124,5 +155,48 @@ public class ScriptLineStepCloseTest {
     assertThatThrownBy(tail::close).isSameAs(firstFailure);
 
     assertThat(closed).containsExactly("line2", "line1", "line0");
+  }
+
+  /**
+   * The walk consumes the run of script lines and then hands whatever the chain ends on back to the ordinary
+   * cascade, so a step of another kind behind the lines is still released.
+   */
+  @Test
+  void aChainEndingOnAnotherKindOfStepStillReleasesThatStep() {
+    final CommandContext context = new BasicCommandContext();
+    final List<String> closed = new ArrayList<>();
+
+    final ScriptLineStep tail = chainOf(context, List.of(
+        new RecordingPlan("line0", closed, null),
+        new RecordingPlan("line1", closed, null)));
+    firstLineOf(tail).setPrevious(new ForeignStep(context, closed, null));
+
+    tail.close();
+
+    assertThat(closed).containsExactly("line1", "line0", "foreign");
+  }
+
+  @Test
+  void aFailureFromThatTrailingStepIsReportedToo() {
+    final CommandContext context = new BasicCommandContext();
+    final List<String> closed = new ArrayList<>();
+    final RuntimeException failure = new IllegalStateException("foreign failed");
+
+    final ScriptLineStep tail = chainOf(context, List.of(
+        new RecordingPlan("line0", closed, null),
+        new RecordingPlan("line1", closed, null)));
+    firstLineOf(tail).setPrevious(new ForeignStep(context, closed, failure));
+
+    assertThatThrownBy(tail::close).isSameAs(failure);
+
+    assertThat(closed).containsExactly("line1", "line0", "foreign");
+  }
+
+  /** Walks back to the head of the script line run so a foreign step can be hung behind it. */
+  private ScriptLineStep firstLineOf(final ScriptLineStep tail) {
+    ScriptLineStep step = tail;
+    while (step.getPrev() instanceof ScriptLineStep previous)
+      step = previous;
+    return step;
   }
 }

